src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,148 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.item = element;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<T>(last, value, null);
+        if (last != null) {
+            last.next = node;
+        }
+        last = node;
+        if (first == null) {
+            first = node;
+        }
```

Comment:
```suggestion
        if (last != null) {
            last.next = node;
        } else {
            first = node;
        }
        last = node;
```

Reply:
it crashes 17 tests, so pls take a look again